include/rtlog/rtlog.h

```diff
@@ -243,7 +257,42 @@ class Logger {
     std::array<char, MaxMessageLength> mMessage{};
   };
 
-  moodycamel::ReaderWriterQueue<InternalLogData> mQueue{MaxNumMessages};
+  class InternalQueue {
+  public:
+    virtual ~InternalQueue() = default;
+    virtual bool tryEnqueue(InternalLogData &&value) = 0;
+    virtual bool tryDequeue(InternalLogData &value) = 0;
+  };
+  class InternalQueueSPSC : public InternalQueue {
+    moodycamel::ReaderWriterQueue<InternalLogData> mQueue{MaxNumMessages};
+
+  public:
+    bool tryEnqueue(InternalLogData &&value) override {
+      return mQueue.try_enqueue(std::move(value));
+    }
+    bool tryDequeue(InternalLogData &value) override {
+      return mQueue.try_dequeue(value);
+    }
+  };
+  class InternalQueueMPSC : public InternalQueue {
+    farbot::fifo<InternalLogData, farbot::fifo_options::concurrency::single,
+                 farbot::fifo_options::concurrency::multiple>
```

Comment:
Smaller comment that we should address after the bigger comments above - let's make the behavior on push/pop explicit:

```
fifo_options::full_empty_failure_mode::return_false_on_full_or_empty,
fifo_options::full_empty_failure_mode::overwrite_or_return_default
```

Are the options. I think both are wait free on write, I guess I would opt for `return_false_on_full_or_empty` so the user can decide what they want to do if the queue fails and data loss happens

Reply:
done.